iceoryx_posh/source/capro/service_description.cpp

```diff
@@ -141,46 +125,32 @@ ServiceDescription::ServiceDescription(const IdString_t& f_service,
     , m_classHash(f_classHash)
     , m_interfaceSource(interfaceSource)
 {
-    if (!(cxx::convert::stringIsNumber(m_serviceString.c_str(), cxx::convert::NumberType::UNSIGNED_INTEGER)
-          && cxx::convert::fromString(m_serviceString.c_str(), m_serviceID)))
-    {
-        m_serviceID = InvalidID;
-    }
-
-    if (!(cxx::convert::stringIsNumber(m_instanceString.c_str(), cxx::convert::NumberType::UNSIGNED_INTEGER)
-          && cxx::convert::fromString(m_instanceString.c_str(), m_instanceID)))
-    {
-        m_instanceID = InvalidID;
-    }
-
-    if (!(cxx::convert::stringIsNumber(m_eventString.c_str(), cxx::convert::NumberType::UNSIGNED_INTEGER)
-          && cxx::convert::fromString(m_eventString.c_str(), m_eventID)))
-    {
-        m_eventID = InvalidID;
-    }
 }
 
 bool ServiceDescription::operator==(const ServiceDescription& rhs) const
 {
-    if ((m_serviceID != AnyService) && (rhs.m_serviceID != AnyService))
+    if ((m_serviceString != IdString_t(cxx::TruncateToCapacity, AnyServiceString))
```

Comment:
Imagining that this logic is always executed I when I search for those gives me the chills. Inefficient, we need to get rid of those wildcards. We can perform the wildcard search differently, as I will propose with the prefix tree approach.

Wildcards are an artifact of the search algorithm, not of the description itself. They should have no bearing on `operator==`. Equality should be simply point-wise (i.e. member-wise) equality here. Even that is inefficient if they are strings but there is no real way around this for now.

Reply:
See [this comment](https://github.com/eclipse-iceoryx/iceoryx/pull/852#discussion_r652905346)